### PR TITLE
Give friendly warning when MySQL runs out of columns

### LINKED ISSUE #3373: Catch and nicely explain ExportToDatabase too many columns error
From [the forum](http://forum.cellprofiler.org/t/cp3-0-error-in-exporttodatabase-during-startup-too-many-columns-on-analysis-per-object/5282) via @PetterRanefall 

>I have a pipeline (based on CellPainting) that works fine in CP2.2, but when I try to run it in CP3.0 I get an error message: “Encoundtered unrecoverable error in ExportToDatabase during startup: too many columns on analysis_Per_Object”.
I am using a SQLite database.

@minh-doan noted:
>While waiting for others’ help, you can try to bypass the error by removing one of the scale at “MeasureTexture” . At the moment it measures both scale 3 and 5. Skip either of them solves the issue.

and that the full error message was
```
Failed to prepare run for module ExportToDatabase
Traceback (most recent call last):
File “/Users/…/CellProfiler/distribution/macos/CellProfiler/cellprofiler/pipeline.py”, line 2085, in prepare_run
if ((not module.prepare_run(workspace)) or
File “/Users/…/CellProfiler/distribution/macos/CellProfiler/cellprofiler/modules/exporttodatabase.py”, line 1836, in prepare_run
self.create_database_tables(self.cursor, workspace)
File “/Users/…/CellProfiler/distribution/macos/CellProfiler/cellprofiler/modules/exporttodatabase.py”, line 2371, in create_database_tables
execute(cursor, statement, return_result=False)
File “/Users/…/CellProfiler/distribution/macos/CellProfiler/cellprofiler/modules/exporttodatabase.py”, line 283, in execute
cursor.execute(query)
OperationalError: too many columns on analysis_Per_Object
```

Pipeline and images attached.
[cellpaintinganalysis.zip](https://github.com/CellProfiler/CellProfiler/files/1447492/cellpaintinganalysis.zip)

### PR DESCRIPTION
**Pull request by AetherUnbound**
Resolves #3373 
Also converts existing settings documentation into the preferred style format

### CODE REVIEW

**Comment by mcquin** [comment]
@AetherUnbound wx should be imported in the methods or functions that require it (unless its a GUI-specific Python module). It's an exception we make so that CellProfiler can run heedless without wx installed. The tests mimic a headless environment.

If you remove the top-level import, and import wx in the methods or functions it's used, Travis should stop complaining.

File: cellprofiler/modules/exporttodatabase.py [hunk 1]
```diff
@@ -1835,13 +2004,22 @@ def prepare_run(self, workspace, as_data_tool=False):
...
                                    if column[0] in onames]
                 self.create_database_tables(self.cursor, workspace)
             return True
+        except sqlite3.OperationalError as err:
+            if err.message.startswith("too many columns"):
+                # Maximum columns reached
+                # https://github.com/CellProfiler/CellProfiler/issues/3373
+                message = "MySQL Error: maximum columns reached. \n" \
+                          "Try exporting a single object per table"
+                cellprofiler.gui.dialog.Error("Error", message)
```
Comment (mcquin) [inline cellprofiler/modules/exporttodatabase.py]: While we're at it, let's create a friendly message for other `OperationalError`s. E.g.,

```python
if err.message.startswith("too many columns"):
    # ...
else:
    cellprofiler.gui.dialog.Error("Error", "MySQL Error: " + err.message)
```

**Comment by AetherUnbound** [comment]
@mcquin it all started by removing the `from cellprofiler.settings import YES, NO`...once I realized the implications of that change and had started modifying the documentation, it was too late to climb back out the rabbit hole :rabbit: 

And no worries! I didn't want the 3 lines I changed that were the really substantive portions of the PR to be lost in all the doc refactoring :)

**Comment by AetherUnbound** [comment]
I also found that the gear icon wasn't loading properly in the documentation - looks like there was a filename mismatch.

File: cellprofiler/modules/exporttodatabase.py [hunk 2]
```diff
@@ -1835,13 +2003,26 @@ def prepare_run(self, workspace, as_data_tool=False):
...
+            if err.message.startswith("too many columns"):
+                # Maximum columns reached
+                # https://github.com/CellProfiler/CellProfiler/issues/3373
+                message = "MySQL Error: maximum columns reached. \n" \
+                          "Try exporting a single object per table. \n\n" \
+                          "Problematic table: {}".format(err.message.replace("too many columns on ", ""))
+            else:
+                # A different MySQL error has occurred, let the user know
+                message = "MySQL Error: {}".format(err.message)
+            cellprofiler.gui.dialog.Error("Error", message)
```
Comment (mcquin) [inline cellprofiler/modules/exporttodatabase.py]: Just raise a `RuntimeError` here (with the friendlier error message). If CellProfiler is running from the GUI, it'll automatically create an error dialog and log a message to the console. If CellProfiler is running headless, it'll only log a message to the console.

**Comment by AetherUnbound** [comment]
FINALLY A PASSING BUILD :tada:

**Pull request merged**